server/mlabns/util/fqdn_rewrite.py

```diff
@@ -0,0 +1,102 @@
+"""Rewrites raw M-Lab FQDNs to apply post-processing or annotations."""
+
+import logging
+
+from mlabns.util import message
+
+
+def rewrite(fqdn, address_family, tool_id):
+    """Rewrites an FQDN to add necessary annotations and special-casing.
+
+    Performs the following rewrites on an FQDN:
+    * Adds a v4/v6 annotation if the client requested an explicit address family
+    * Applies a workaround to fix FQDNs for NDT-SSL queries.
+
+    Args:
+        fqdn: A tool FQDN with no address family specific annotation.
+        address_family: The address family for which to create the FQDN or None
+            to create an address family agnostic FQDN.
+        tool_id: Name of tool associated with the FQDN (e.g. 'ndt_ssl').
+
+    Returns:
+        FQDN after rewriting to apply all modifications to the raw FQDN.
+    """
+    rewritten_fqdn = _apply_af_awareness(fqdn, address_family)
```

Comment:
NOTE: AF domain name decorations are _not_ currently implemented nor published.  We _don't_ want to decorate domain names for NDT-SSL with AF until or if we actually publish such records.   @stephen-soltesz may have more insight into why since he implemented the change to create the "flat" names.  I suspect it's because we don't currently really run NDT over IPv6.  However, I believe we did enable IPv6 for [a select couple nodes](https://github.com/m-lab/operator/commit/523e54a38296d9180d04f1af48a7470b0e1ef48a) 5 or 6 months ago, specifically nuq02 and lga06.